src/bp-xprofile/classes/class-bp-xprofile-profiledata.php

```diff
@@ -126,15 +126,19 @@ public function populate( $field_id, $user_id ) {
 	public function exists() {
 		global $wpdb;
 
-		// Check cache first.
 		$cache_key = "{$this->user_id}:{$this->field_id}";
 		$cached    = wp_cache_get( $cache_key, 'bp_xprofile_data' );
+		$retval    = false;
 
 		if ( $cached && ! empty( $cached->id ) ) {
 			$retval = true;
 		} else {
 			$bp     = buddypress();
 			$retval = $wpdb->get_row( $wpdb->prepare( "SELECT id FROM {$bp->profile->table_name_data} WHERE user_id = %d AND field_id = %d", $this->user_id, $this->field_id ) );
+
+			if ( ! is_wp_error( $retval ) && ! empty( $retval ) ) {
```

Comment:
Also here, I think you want 
`if ( ! is_wp_error( $dbval ) && ! empty( $dbval ) ) {`
so you're only setting `$retval` to true (at line 140) when the database returns a valid value.